### PR TITLE
Combine spec and X-Ray backend attributes into python template

### LINKED ISSUE #170: Add back RPC attributes to Python expected template
We should combine the attributes recorded in #165 and #168 so that we are verifying *both* `aws.operation` is set on an inferred segment and the three `rpc` attributes are set on the span per the spec. We can do this once the translation is updated in the awsxrayexporter and we've upgraded the Python Lambda Layer to include this Collector Contrib `awsxrayexporter` change.

### PR DESCRIPTION
**Pull request by NathanielRN**
**Description:**

Similar to https://github.com/aws-observability/aws-otel-test-framework/pull/383, this PR is meant to update the template to incorporate BOTH [the spec defined attributes](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/instrumentation/aws-sdk.md#common-attributes) and the attributes we need the Collector to set for the X-Ray backend to read https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/6109

**Merge once** https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/6109 is FIXED.

**Link to tracking Issue:**

Fixes #170

**Testing:**

Read more about why we know these are the changes we need in https://github.com/aws-observability/aws-otel-test-framework/pull/383. Once this merges the `main-build.yml` for Python **should start passing again**.

**Documentation:**

N/A

### CODE REVIEW

File: adot/utils/expected-templates/python.json [hunk 1]
```diff
@@ -9,7 +9,27 @@
...
+          "name": "HTTP GET"
+      },
+      {
+          "name": "S3",
+          "aws": {
+            "operation": "ListBuckets"
+          },
+          "metadata": {
+            "default": {
+              "aws.service": "s3",
```
Comment (NathanielRN) [inline adot/utils/expected-templates/python.json:24]: Similar to https://github.com/aws-observability/aws-otel-test-framework/pull/383#discussion_r742113179, not sure if this is needed. Maybe it is used by the AWS X-Ray backend to get the `"origin":"AWS::S3",` value on the S3 inferred span?

File: adot/utils/expected-templates/python.json [hunk 2]
```diff
@@ -9,7 +9,27 @@
...
+              "rpc.service": "S3",
```
Comment (anuraaga) [inline adot/utils/expected-templates/python.json:25]: Since X-Ray backend doesn't use these fields, I'd recommend not actually validating them

**Review by anuraaga — COMMENTED** [review]
One general thing to keep in mind is this test framework verifies the X-Ray data model, not the OTel data model. So we probably don't need OTel attributes in it - in particular, it prevents removing `rpc.service` / `rpc.method` in the xray exporter. We probably should remove them instead of sending redundant information (for most other attributes such as http.status_code, we do remove them when populating an X-Ray "native" data model field).

File: adot/utils/expected-templates/python.json (hunk 2 above)
Comment (NathanielRN) [reply adot/utils/expected-templates/python.json]: Thanks for the call out! It makes great sense to me 🙂 I've made the update!

**Comment by NathanielRN** [comment]
Blocked on LambdaComponents `v0.15.0` release.

**Comment by NathanielRN** [comment]
[Most recent Canary CI](https://github.com/aws-observability/aws-otel-lambda/runs/4309440750?check_suite_focus=true#step:21:54) shows that `[3].subsegments[1].aws.operation` is correctly set to "ListBuckets".

**Review by willarmiros — APPROVED** [review]
🎉

**Pull request merged**